### PR TITLE
Introduces a parameter to limit the allowed maximum spatial refinement level

### PR DESCRIPTION
**Pull request by ykempf**
Up for discussion and CI.

### CODE REVIEW

**Comment by ykempf** [comment]
Eh. Let me change that already.

**Comment by ykempf** [comment]
So the terminology of the new flag is: we have the already-existing max refinement level and I don't want to rename this, and we have the new one that tells what the actually allowed maximum is, and that should be smaller than the other pre-existing flag. But I see how this _allowed_ terminology might be confusing, so suggestions welcome to make this better. And the next push will allow for not setting the new flag in configs so the users don't have surprises...

File: parameters.cpp [hunk 1]
```diff
@@ -157,6 +157,7 @@ Realf P::vamrCoarsenLimit = 0.5;
 string P::vamrVelRefCriterion = string("");
 
 uint P::amrMaxSpatialRefLevel = 0;
+uint P::amrMaxAllowedSpatialRefLevel = 0;
```
Comment (lkotipal) [inline parameters.cpp:160]: Rather have this as a signed integer, with e.g. negative values for automatic behavior

Comment (ykempf) [reply parameters.cpp:160]: Ah, *this* is why it wasn't merged yet. :)

Comment (ykempf) [reply parameters.cpp]: Better?

Comment (lkotipal) [reply parameters.cpp]: Perfect

**Comment by ykempf** [comment]
Yay! A pass from CI! Merge?

**Pull request merged**